src/axom/primal/geometry/BezierPatch.hpp

```diff
@@ -0,0 +1,875 @@
+
+  /// Returns a copy of the Bezier patch's weights
+  axom::Array<T, 2> getWeights() const { return m_weights; }
+
+  /*!
+   * \brief Reverses the order of one direction of the Bezier patch's control points and weights
+   *
+   * \param [in] axis orientation of patch. 0 to reverse in u, 1 for reverse in v
+   */
+  void reverseOrientation(int axis)
+  {
+    const int ord_u = getOrder_u();
+    const int mid_u = (ord_u + 1) / 2;
+
+    const int ord_v = getOrder_v();
+    const int mid_v = (ord_v + 1) / 2;
+
+    if(axis == 0)
+    {
+      for(int q = 0; q <= ord_v; ++q)
+      {
+        for(int i = 0; i < mid_u; ++i)
+          axom::utilities::swap(m_controlPoints(i, q),
+                                m_controlPoints(ord_u - i, q));
+
+        if(isRational())
+          for(int i = 0; i < mid_u; ++i)
+            axom::utilities::swap(m_weights(i, q), m_weights(ord_u - i, q));
+      }
+    }
+    else
+    {
+      for(int p = 0; p <= ord_u; ++p)
+      {
+        for(int i = 0; i < mid_v; ++i)
+          axom::utilities::swap(m_controlPoints(p, i),
+                                m_controlPoints(p, ord_v - i));
+
+        if(isRational())
+          for(int i = 0; i < mid_v; ++i)
+            axom::utilities::swap(m_weights(p, i), m_weights(p, ord_v - i));
+      }
+    }
+  }
+
+  /// Swap the axes such that s(u, v) becomes s(v, u)
+  void swapAxes()
+  {
+    const int ord_u = getOrder_u();
+    const int ord_v = getOrder_v();
+
+    CoordsMat new_controlPoints(ord_v + 1, ord_u + 1);
+
+    for(int p = 0; p <= ord_u; ++p)
+      for(int q = 0; q <= ord_v; ++q)
+        new_controlPoints(q, p) = m_controlPoints(p, q);
+
+    m_controlPoints = new_controlPoints;
+
+    if(isRational())
+    {
+      axom::Array<T, 2> new_weights(ord_v + 1, ord_u + 1);
+      for(int p = 0; p <= ord_u; ++p)
+        for(int q = 0; q <= ord_v; ++q) new_weights(q, p) = m_weights(p, q);
+
+      m_weights = new_weights;
+    }
+  }
+
+  /// Returns an axis-aligned bounding box containing the Bezier patch
+  BoundingBoxType boundingBox() const
+  {
+    return BoundingBoxType(m_controlPoints.data(),
+                           static_cast<int>(m_controlPoints.size()));
+  }
+
+  /// Returns an oriented bounding box containing the Bezier patch
+  OrientedBoundingBoxType orientedBoundingBox() const
+  {
+    return OrientedBoundingBoxType(m_controlPoints.data(),
+                                   static_cast<int>(m_controlPoints.size()));
+  }
+
+  /*!
+   * \brief Evaluates a slice Bezier patch for a fixed parameter value of \a u or \a v
+   *
+   * \param [in] u parameter value at which to evaluate the first axis
+   * \param [in] v parameter value at which to evaluate the second axis
+   * \param [in] axis orientation of curve. 0 for fixed u, 1 for fixed v
+   * \return p the value of the Bezier patch at (u, v)
+   *
+   * \note We typically evaluate the patch at \a u or \a v between 0 and 1
+   */
+  BezierCurveType isocurve(T uv, int axis = 0) const
```

Comment:
Thought: Would this class be less error prone w/ an `Axis` enum.

E.g.
```cpp
class BezierPatch
{  
   enum class Axis {U=0, V=1};
   ...
   BezierCurveType isocurve(T uv, Axis axis = Axis::U) const;
};
```

Reply:
I don't really think so. The `axis=0` vs `axis=1` distinction is pretty commonly used by Python and Matlab, so I don't think it's all that unintuitive at the moment. As far as bugs go, you end up just switching from one one-character difference (0/1) to a different one (u/v), so I don't think it would be worth it in that respect either.

Reply:
Thanks @jcs15c -- I'm ok w/ the integer if people are already used to accessing it that way

Reply:
I went through and added `_u` and `_v` versions of each of the relevant methods (in addition to the existing versions), which I think will add the missing clarity that just an axis parameter lacks.